### PR TITLE
Dpys

### PR DESCRIPTION
**Pull request by dPys**
*Tweak DSN params
*Increase seeding density to 20
*Add graspy as a dependency to requirements.txt and Dockerfile
*Swap graph plotting from nilearn with graph plotting from graspy

### CODE REVIEW

File: ndmg/register/gen_reg.py [hunk 1]
```diff
@@ -115,23 +95,37 @@ def direct_streamline_norm(streams, fa_path, namer):
...
     mgu.show_template_bundles(mni_streamlines, template_path, streams_warp_png)
 
-    return mni_streamlines
+    return mni_streamlines, streams_mni
 
 
 class dmri_reg(object):
     def __init__(self, namer, nodif_B0, nodif_B0_mask, t1w_in, vox_size, simple):
+        import os.path as op
+        if os.path.isdir("/ndmg_atlases"):
```
Comment (loftusa) [inline ndmg/register/gen_reg.py:113]: maybe we should just make a `get_atlas_dir` function?

File: ndmg/graph/gen_graph.py [hunk 2]
```diff
@@ -290,14 +296,12 @@ def save_graph_png(self, graphname):
...
+        from graspy.plot import heatmap
 
-        from sklearn.preprocessing import normalize
         conn_matrix = np.array(nx.to_numpy_matrix(self.g))
-        conn_matrix = normalize(conn_matrix)
-        [z_min, z_max] = -np.abs(conn_matrix).max(), np.abs(conn_matrix).max()
-        plot_matrix(conn_matrix, figure=(10, 10), vmax=z_max * 0.5, vmin=z_min * 0.5, auto_fit=True, grid=False,
-                    colorbar=False)
+        conn_matrix = ptr.pass_to_ranks(conn_matrix)
+        heatmap(conn_matrix)
```
Comment (loftusa) [inline ndmg/graph/gen_graph.py:304]: I think we still need to normalize plotting (e.g., vmin and vmax are 0 and 1, respectively)

File: Dockerfile [hunk 3]
```diff
@@ -63,7 +63,7 @@ RUN \
     pip3.6 install numpy
 
 RUN \
-    pip3.6 install nibabel dipy scipy python-dateutil pandas boto3 awscli matplotlib nilearn sklearn pandas cython vtk pyvtk fury awscli requests ipython duecredit
+    pip3.6 install nibabel dipy scipy python-dateutil pandas boto3 awscli matplotlib nilearn sklearn pandas cython vtk pyvtk fury awscli requests ipython duecredit graspy
```
Comment (loftusa) [inline Dockerfile:66]: yeet

**Review by loftusa — APPROVED** [review]
Looks good. I'd add value normalization to heatmap, but it's a one-liner so I can just do that myself later.

Approved

File: ndmg/graph/gen_graph.py [hunk 4]
```diff
@@ -290,14 +296,12 @@ def save_graph_png(self, graphname):
         import matplotlib
         matplotlib.use('agg')
         from matplotlib import pyplot as plt
-        from nilearn.plotting import plot_matrix
+        from graspy.utils import ptr
+        from graspy.plot import heatmap
 
-        from sklearn.preprocessing import normalize
         conn_matrix = np.array(nx.to_numpy_matrix(self.g))
```
Comment (loftusa) [inline ndmg/graph/gen_graph.py:302]: I'm wondering if this is where the sorting discrepancy between our plots happens. I'm using graspy's [`import_edgelist`](https://github.com/neurodata/graspy/blob/master/graspy/utils/utils.py#L152), which contains these lines:

```
    vertices = np.sort(reduce(np.union1d, [G.nodes for G in graphs]))
    out = [nx.to_numpy_array(G, nodelist=vertices, dtype=np.float) for G in graphs]
```
this sorts by vertex number, and adds nodes to account for dropped nodes.

Comment (dPys) [reply ndmg/graph/gen_graph.py:302]: Interesting.

Here's an alternative way:
```
import networkx as nx
import numpy as np

edgelist_path = '/tmp/output_desikan_det_local_csd_native/dwi/roi-connectomes/desikan_space-MNI152NLin6_res-2x2x2/sub-0025427_ses-1_dwi_desikan_space-MNI152NLin6_res-2x2x2_measure-spatial-ds_adj.ssv'

g_arr = np.loadtxt(open(edgelist_path, "rb"), delimiter=" ")

G = nx.Graph()

for i in g_arr:
    [edge_i, edge_j, weight] = i
    G.add_edges_from([(edge_i, edge_j)], weight=weight)

conn_matrix = nx.to_numpy_matrix(G)
```

File: ndmg/register/gen_reg.py (hunk 1 above)
Comment (dPys) [reply ndmg/register/gen_reg.py:113]: I love that idea. Feel free to create a prototype!

File: ndmg/graph/gen_graph.py (hunk 2 above)
Comment (dPys) [reply ndmg/graph/gen_graph.py:304]: Actually, if you look through the graspy code for heatmap, you will see that vmin/vmax are inferred based on the data if these parameters are not set explicitly. @j1c can correct me if I'm wrong this.

File: ndmg/register/gen_reg.py (hunk 1 above)
Comment (loftusa) [reply ndmg/register/gen_reg.py:113]: > I love that idea. Feel free to create a prototype!

will put it on my growing list of minor stuff to do! (although, pretty low-priority I think)

File: ndmg/graph/gen_graph.py (hunk 2 above)
Comment (loftusa) [reply ndmg/graph/gen_graph.py:304]: > Actually, if you look through the graspy code for heatmap, you will see that vmin/vmax are inferred based on the data if these parameters are not set explicitly. @j1c can correct me if I'm wrong this.

Yes, that's true - the reason we need to hardcore these parameters is so that we can compare multiple plots. If I have 2 heatmaps, and vmax is `1` on the first, and `10` on the second, and the underlying data are the same, the heatmaps will look completely different.

Comment (j1c) [reply ndmg/graph/gen_graph.py:304]: `vmax` and `vmin` just clips the data. if `vmax` is 1, then all values above 1 is set to 1. 

This is why i think it makes sense to at least PTR prior to plotting because then values are always between 0 and 1 so no need to mess with `vmax` and `vmin`, and makes all the small values visible.

File: ndmg/graph/gen_graph.py (hunk 4 above)
Comment (j1c) [reply ndmg/graph/gen_graph.py:302]: We give an arbitrary sorting to the nodes because the way you make the graph here doesn't guarantee that the first element of the `conn_matrix` correspond to the same nodes across multiple edgelists. `nx.to_numpy_matrix()` just builds the array based on the order of the nodes that `g_arr` is in, which is arbitrary node ordering.

File: ndmg/graph/gen_graph.py (hunk 2 above)
Comment (j1c) [reply ndmg/graph/gen_graph.py:304]: Also graspy heatmap function also allow you to just pass in a `transform` argument that will take care of PTR for you so no need to PTR separately before plotting.

Comment (dPys) [reply ndmg/graph/gen_graph.py:304]: Thanks @j1c . Thought so!

File: ndmg/graph/gen_graph.py (hunk 4 above)
Comment (dPys) [reply ndmg/graph/gen_graph.py:302]: Good to know!

File: ndmg/graph/gen_graph.py (hunk 2 above)
Comment (loftusa) [reply ndmg/graph/gen_graph.py:304]: Ok, thanks dude
